app/src/main/java/org/oppia/android/app/help/thirdparty/LicenseListViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package org.oppia.android.app.help.thirdparty
+
+import android.annotation.SuppressLint
+import androidx.appcompat.app.AppCompatActivity
+import org.oppia.android.R
+import org.oppia.android.app.viewmodel.ObservableViewModel
+import javax.inject.Inject
+
+/** View model in [LicenseListFragment]. */
+class LicenseListViewModel @Inject constructor(
+  val activity: AppCompatActivity,
+  val index: Int
+) : ObservableViewModel() {
+  private val arrayList = ArrayList<LicenseItemViewModel>()
+
+  /** Stores the list of third-party dependencies. */
+  val licenseItemList: List<LicenseItemViewModel> by lazy {
+    getRecyclerViewItemList()
+  }
+
+  @SuppressLint("ResourceType")
+  private fun getRecyclerViewItemList(): ArrayList<LicenseItemViewModel> {
+    val licenses =
+      activity.resources.obtainTypedArray(R.array.third_party_dependency_license_names_array)
+    val stringArrayResId = licenses.getResourceId(index, -1)
+    val licenseNames = activity.resources.getStringArray(stringArrayResId)
+
+    licenseNames.forEachIndexed { licenseIndex, name ->
+      val licenseItemViewModel =
+        LicenseItemViewModel(activity, name, licenseIndex, index)
+      arrayList.add(licenseItemViewModel)
+    }
+    licenses.recycle()
```

Comment:
Do we actually need this? I am not sure about this.

Reply:
I saw it here - https://stackoverflow.com/a/5931094/12215015
```java
Resources res = getResources();
TypedArray ta = res.obtainTypedArray(R.array.array0);
int n = ta.length();
String[][] array = new String[n][];
for (int i = 0; i < n; ++i) {
    int id = ta.getResourceId(i, 0);
    if (id > 0) {
        array[i] = res.getStringArray(id);
    } else {
        // something wrong with the XML
    }
}
ta.recycle(); // Important!
```

Reason hasn't been mentioned here.

Reply:
Ack.